### PR TITLE
Create consumer using rack id

### PR DESCRIPTION
**Pull request by roy-dydx**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: indexer/packages/kafka/src/consumer.ts [hunk 1]
```diff
@@ -51,38 +47,51 @@ export function updateOnBatchFunction(
...
 
-export async function stopConsumer(): Promise<void> {
-  logger.info({
-    at: 'kafka-consumer#stop',
-    message: 'Stopping kafka consumer',
-    groupId,
+    if (!stopped) {
+      await consumer!.connect();
+      logger.info({
+        at: 'kafka-consumer#disconnect',
```
Comment (dydxwill) [inline indexer/packages/kafka/src/consumer.ts:83]: kafka-consumer#initConsumer?

File: indexer/packages/kafka/src/consumer.ts [hunk 2]
```diff
@@ -51,38 +47,51 @@ export function updateOnBatchFunction(
...
+        message: 'Kafka consumer reconnected',
+        groupId,
+      });
+    } else {
+      logger.info({
+        at: 'kafka-consumer#disconnect',
```
Comment (dydxwill) [inline indexer/packages/kafka/src/consumer.ts:89]: kafka-consumer#initConsumer?

File: indexer/packages/kafka/src/consumer.ts [hunk 3]
```diff
@@ -51,38 +47,51 @@ export function updateOnBatchFunction(
...
+    readUncommitted: false,
+    maxBytes: 4194304, // 4MB
+    rackId: await getAvailabilityZoneId(),
+  });
+
+  consumer!.on('consumer.disconnect', async () => {
     logger.info({
-      at: 'kafka-consumer#disconnect',
-      message: 'Not reconnecting since task is shutting down',
+      at: 'consumers#disconnect',
```
Comment (dydxwill) [inline indexer/packages/kafka/src/consumer.ts:75]: kafka-consumer#initConsumer?

**Review by dydxwill — APPROVED** [review]

File: indexer/packages/kafka/src/consumer.ts (hunk 1 above)
Comment (roy-dydx) [reply indexer/packages/kafka/src/consumer.ts:83]: It's `.on()`?

**Pull request merged**